mods/env_sounds/init.lua

```diff
@@ -0,0 +1,97 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:get_pos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
+	-- Total number of waters in radius
+	local waters = #wpos
+	if waters == 0 then
+		return
+	end
+
+	-- Find average position of water positions
+	local wposav = vector.new()
+	for i, pos in ipairs(wpos) do
```

Comment:
That `ipairs` is not needed:
```suggestion
	for i = 1, waters do
		local pos = wpos[i]
```
At least use `_` instead of `i`:
```suggestion
	for _, pos in ipairs(wpos) do
```

Reply:
What are the advantages of your suggestions?

Reply:
The first suggestion is probably mostly a matter of taste (sorry).

It is common practice that `_` is used for not used variables.

Reply:
Hey no problem =)
I'll do the 2nd suggestion.

Reply:
I'm guessing that discarding the unused value has a small performance benefit as it isn't stored.

Reply:
> I'm guessing that discarding the unused value has a small performance benefit as it isn't stored.

FYI, `_` is just a normal variable, it isn't treated differently.